packages/react-components/src/editor/editor.tsx

```diff
@@ -0,0 +1,240 @@
+  ) => void;
+  editable?: boolean;
+  children?: ReactNode;
+  showToolbar?: boolean;
+  toolbarTools?: ToolbarTools;
+  showFloatingToolbar?: boolean;
+  nodes?: InitialConfigType['nodes'];
+  floatingToolbarTools?: FloatingToolbarTools;
+  features?: (keyof typeof featuresToNodeMapping)[];
+  dataManagementRef?: RefObject<DataManagementPluginHandle>;
+}) => {
+  const supportedFeatures = useMemo(() => {
+    if (features) {
+      const selectedNodes = features
+        .map((item) => featuresToNodeMapping[item].node)
+        .flat();
+
+      const selectedPlugins = features
+        .map((item) => featuresToNodeMapping[item].component)
+        .flat()
+        .map((item) => ({ component: item, id: featureId++ }));
+
+      return { nodes: selectedNodes, components: selectedPlugins };
+    }
+
+    const defaultFeatures = {
+      nodes: Object.values(featuresToNodeMapping)
+        .map((item) => item.node)
+        .flat(),
+      components: Object.values(featuresToNodeMapping)
+        .map((item) => item.component)
+        .flat()
+        .map((item) => ({ component: item, id: featureId++ }))
+    };
+
+    return defaultFeatures;
+  }, [features]);
+
+  const customFeatures = useMemo(() => {
+    const featuresContainer = findReactChildByType(
+      children,
+      EditorFeatures
+    ) as ReactElement;
+
+    const customFeatureNodes: ReadonlyArray<Klass<LexicalNode>>[] = [];
+    const customFeatureElements: ReactElement[] = [];
+
+    if (featuresContainer) {
+      findReactChildrenByType(
+        featuresContainer.props.children,
+        EditorFeatures.Feature
+      ).forEach((_child) => {
+        if (
+          _child
+          && isValidElement(_child)
+          && _child.type === EditorFeatures.Feature
+        ) {
+          const child = _child as ReactElement<EditorFeatureProps>;
+          customFeatureElements.push(child);
+          if (child.props.node) {
+            if (Array.isArray(child.props.node)) {
+              customFeatureNodes.push(...child.props.node);
+            } else {
+              customFeatureNodes.push(child.props.node);
+            }
+          }
+        }
+      });
+    }
+
+    return {
+      nodes: customFeatureNodes.flat(),
+      elements: customFeatureElements.map((item) => ({
+        component: item,
+        id: customFeatureId++
+      }))
+    };
+  }, [children]);
+
+  const customContent = useMemo(() => {
+    const content = findReactChildByType(
+      children,
+      EditorFeatures.Content
+    ) as ReactElement;
+
+    if (content) {
+      return content;
+    }
+    return null;
+  }, [children]);
+
+  return (
+    <LexicalComposer
+      initialConfig={{
+        ...initialConfig,
+        editorState,
+        editable,
+        nodes: [
+          ...supportedFeatures.nodes,
+          ...customFeatures.nodes,
+          ...(nodes ?? [])
+        ]
+      }}
+    >
+      <>{showToolbar && <ToolbarPlugin tools={toolbarTools} />}</>
+      <EditorArea />
```

Comment:
how do we style EditorArea? Using `class: editor-container ` ?

Reply:
Yes, currently we will have to use class name to target the element